src/custom/components/HighFeeWarning/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useContext, useMemo } from 'react'
+import { AlertTriangle } from 'react-feather'
+import styled, { ThemeContext } from 'styled-components'
+import { Fraction } from '@uniswap/sdk-core'
+import { MouseoverTooltipContent } from 'components/Tooltip'
+import { StyledInfo } from 'pages/Swap/styleds'
+import { useHighFeeWarning } from 'state/swap/hooks'
+import TradeGp from 'state/swap/TradeGp'
+import { AuxInformationContainer } from '../CurrencyInputPanel'
+import { darken } from 'polished'
+
+interface HighFeeContainerProps {
+  padding?: string
+  margin?: string
+  width?: string
+  bgColour?: string
+  textColour?: string
+}
+
+const WarningCheckboxContainer = styled.div`
+  display: flex;
+  font-weight: bold;
+  gap: 2px;
+  justify-content: center;
+  align-items: center;
+`
+
+const HighFeeWarningContainer = styled(AuxInformationContainer).attrs((props) => ({
+  ...props,
+  hideInput: true,
+}))<HighFeeContainerProps>`
+  &&&&& {
+    background: ${({ theme, bgColour }) => bgColour || theme.info};
+    color: ${({ theme, textColour }) => textColour || theme.infoText};
+  }
+  padding: ${({ padding = '5px 12px' }) => padding};
+  width: ${({ width = '100%' }) => width};
+  border-radius: ${({ theme }) => theme.buttonPrimary.borderRadius};
+  margin: ${({ margin = '0 auto 12px auto' }) => margin};
+
+  > div {
+    display: flex;
+    justify-content: space-evenly;
+    align-items: center;
+    gap: 5px;
+
+    font-size: 13px;
+    font-weight: 500;
+
+    svg {
+      &:first-child {
+        margin-right: 5px;
+      }
+      stroke: ${({ theme, textColour }) => textColour || theme.infoText};
+    }
+
+    > ${WarningCheckboxContainer} {
+      font-size: 10px;
+      margin-left: auto;
+      min-width: max-content;
+
+      > input {
+        cursor: pointer;
+        margin: 1px 4px 0 0;
+      }
+    }
+  }
+`
+
+const ErrorStyledInfo = styled(StyledInfo)`
+  color: ${({ theme }) => theme.infoText};
+`
+const HIGH_TIER_FEE = { level: '30', colour: '#FF7676' }
```

Comment:
Could consider defining theme colors and pull from there (as we can tweak/re-use the colors).